roles/neutron-data-network/tasks/main.yml

```diff
@@ -167,22 +167,26 @@
     - haproxy.loadbalancer.j2
     - haproxy_proxies.j2
   notify: restart neutron lbaas agent
-  when: (neutron.lbaas.enabled == "smart" and
+  when: (os != 'rhel' or openstack_install_method != 'distro') and ((neutron.lbaas.enabled == "smart" and
```

Comment:
is this needed for openstack_install_method == 'source'  (i.e ubuntu and centos ?) or is needed for source ((ubuntu, centos) and package (ubuntu giftwrap)?

if is latter than might want to have only openstack_install_method != 'distro'  in 1st part of the when clause